Fix temperature breach filter

mark-prins opened the pull request:
<!-- IMPORTANT!
  - Every PR must reference an issue; this helps to explain the intent of the PR
 -->

Fixes #2537

# 👩🏻‍💻 What does this PR do? 
 <!-- Explain the changes you made, and why they're needed. Add a screenshot if you've made any UI changes!  -->
Could not repro the stated issue. A related issue was raised in conversation - that the breaches aren't showing.

That's the part which I've addressed here - it turns out that the breach filter was a little over complicated and didn't function as required. The filter was looking for `temperature_breach` records.. the query was

```
SELECT "temperature_log"."id", "temperature_log"."temperature", "temperature_log"."sensor_id", "temperature_log"."location_id", "temperature_log"."store_id", "temperature_log"."datetime", "temperature_log"."temperature_breach_id" 
FROM "temperature_log" 
WHERE "temperature_log"."temperature_breach_id" IN 
(SELECT "temperature_breach"."id" FROM "temperature_breach" WHERE "temperature_breach"."id" = 'fff4bac7-2256-43cf-acb9-78acfc8cc7e0') 
ORDER BY "temperature_log"."datetime" DESC
```

which will not work for new breaches, the breach has to be added before the breach can be added!

I've changed the filter to look for an `id` only, which simplifies things and allows breaches to be added.

# 🧪 How has/should this change been tested? 
<!-- Explain how to setup for testing here if it is not already obvious, and how you've tested this PR. -->
Configure a CCA with a sensor and sync with omSupply. get the sensor into a breach and check if the breach shows in oms.

## 💌 Any notes for the reviewer?
<!-- eg. Do you have any specific questions for the reviewer? Is there a high risk/complicated change they should focus on? If there are any general areas of the codebase your changes might have have touched or could cause side effects to, mention them here.-->

## 📃 Documentation
<!-- Note down any areas which require documentation updates -->
_No user facing changes_

mark-prins commented on client/packages/coldchain/src/Monitoring/ListView/TemperatureChart/TemperatureChart.tsx:
```diff
@@ -177,6 +177,7 @@ const Chart = ({
                       borderStyle: 'solid',
                       borderColor: theme.palette.gray.light,
                       padding: 3,
+                      textAlign: 'left',
```
Offroad: the legend was showing with names centred:

<img width="273" alt="Screenshot 2023-12-06 at 10 24 39 AM" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/b581b61a-7eac-40fb-87fc-e2be71aa98fe">

mark-prins commented on server/repository/src/db_diesel/temperature_log.rs:
```diff
@@ -92,6 +90,9 @@ impl<'a> TemperatureLogRepository<'a> {
             query = query.order(temperature_log_dsl::datetime.desc())
         }
 
+        // Debug diesel query
```
just to help with debugging 🤷

mark-prins commented on server/server/src/cold_chain/sensor.rs:
```diff
@@ -109,7 +109,7 @@ fn validate_sensor(sensor: &Sensor) -> Result<(), String> {
     match sensor.log_delay {
         Some(log_delay) => {
-            if log_delay < 1 {
+            if log_delay < 0 {
```
the 'add a debug sensor' function creates a record with `0` here, which meant that the sensor mutation fails for all sensors. since positive number is 0 and up, I figured changing this test still fits the intention

mark-prins replied:
in the cold chain app - under developer settings you can add a sensor, and then add random logs for it.

andreievg commented on server/server/src/cold_chain/temperature_breach.rs:
```diff
@@ -126,11 +126,7 @@ fn upsert_temperature_breach(
-                    TemperatureBreachFilter::new().id(EqualFilter::equal_to(&id)),
-                ),
-            ),
+            Some(TemperatureLogFilter::new().temperature_breach_id(EqualFilter::equal_to(&id))),
```
I forgot, why do we need to check that there are logs for the breach ?

andreievg commented on server/graphql/types/src/types/temperature_log.rs:
```diff
@@ -42,7 +41,7 @@ pub struct TemperatureLogFilterInput {
     pub sensor: Option<SensorFilterInput>,
     pub location: Option<LocationFilterInput>,
-    pub temperature_breach: Option<TemperatureBreachFilterInput>,
+    pub temperature_breach_id: Option<EqualFilterStringInput>,
```
I think the breach type filter might be broken with this change ?

andreievg reviewed (changes requested):
@mark-prins.

I haven't tested, but I can't see the difference between the queries, also there is a comment about the breach filter.

The reason I can't see the difference, is there is a referencial contraint on logs, can't set temperature_breach_id, unless the breach exists, so I can't quite see what the difference is between the new filter and the old. 

Btw does cold chain app push temperature_logs first or breaches first ? If it's the former then we might have an issue with referencial contraints.

Unforntunately I don't have the cold chain app set up, I wonder if we can share sqlite filter for it, or if there are some postman queries we can use to replicate/validate ?

mark-prins commented:
Thanks Andrei for review.
I've reverted the filter changes, and removed the log check. Now the api will accept breaches which don't have a log in the database.

To remove any UI strangeness I've added a check in the notification component - if there's no min/max temperature then it just doesn't show. The temperature is also blank in the breach table.

mark-prins commented:
Oh, and I'm using postgres, so don't have a sqlite file to share. You can use this postman collection to add a breach or log. I've been using CCA with some sensors that I'm shuttling in and out of the fridge to get it to breach which is a bit of a pain, but it did pick up this particular error that I'd missed earlier.

[Cold Chain.postman_collection.json.zip](https://github.com/msupply-foundation/open-msupply/files/13595278/Cold.Chain.postman_collection.json.zip)

andreievg reviewed (approved):
Looks good, thanks for changes @mark-prins.
